# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ProcessPriorityManager.h
# Commit: 07c8392cb04d
# Full Hash: 07c8392cb04d095caa8e65ee1a10ccc5f7352eb6
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2021-05-07 21:46:25
# Regressor Bug: 1618547
# File Overlap Count: 1
# Description:
#   Bug 1618547 - Support Fission in the process priority manager. r=kmag,gsvelto
#   
#   With Fission, there can be multiple BrowserParents in a single tab, so
#   this patch moves the tracking of active tabs onto the top BrowsingContext
#   in a tab. If the priority of a top BC is changed, then the activity
# ==============================================================================

diff -r 285861b32934 -r 07c8392cb04d dom/ipc/ProcessPriorityManager.h
--- a/dom/ipc/ProcessPriorityManager.h	Fri May 07 17:42:21 2021 +0000
+++ b/dom/ipc/ProcessPriorityManager.h	Fri May 07 17:51:34 2021 +0000
@@ -14,6 +14,7 @@
 namespace mozilla {
 namespace dom {
 class BrowserParent;
+class CanonicalBrowsingContext;
 class ContentParent;
 }  // namespace dom
 
@@ -70,6 +71,8 @@
    */
   static bool CurrentProcessIsForeground();
 
+  static void ActivityChanged(dom::CanonicalBrowsingContext* aBC,
+                              bool aIsActive);
   static void ActivityChanged(dom::BrowserParent* aBrowserParent,
                               bool aIsActive);
 